libs/components/src/lib/header/header.spec.ts

```diff
@@ -0,0 +1,121 @@
+import { elementUpdated, fixture, getControlElement } from '@vivid-nx/shared';
+import { header } from './header';
+import '.';
+
+const COMPONENT_TAG = 'vwc-header';
+
+describe('vwc-header', () => {
+	let element: header;
+
+	beforeEach(async () => {
+		element = (await fixture(
+			`<${COMPONENT_TAG}></${COMPONENT_TAG}>`
+		)) as header;
+	});
+
+	describe('basic', () => {
+		it('should be initialized as a vwc-header', async () => {
+			expect(element).toBeInstanceOf(header);
+			expect(element.alternate).toBeFalsy();
+			expect(element.fixed).toBeFalsy();
+			expect(element.heading).toBeUndefined();
+		});
+	});
+
+	describe('fixed', () => {
+		it('should set "fixed" to true and add "fixed" class', async () => {
+			const control = getControlElement(element);
+			let hasClassFixed = control.classList.contains('fixed');
+			element.fixed = true;
+			element.connectedCallback();
+			await elementUpdated(element);
+			expect(hasClassFixed).toEqual(false);
+
+			hasClassFixed = control.classList.contains('fixed');
+			expect(hasClassFixed).toEqual(true);
+		});
+	});
+
+	describe('alternate', () => {
+		it('should set "alternate" to true and add "alternate" class', async () => {
+			const control = getControlElement(element);
+			let hasClassAlternate = control.classList.contains('alternate');
+			element.alternate = true;
+			await elementUpdated(element);
+			expect(hasClassAlternate).toEqual(false);
+
+			hasClassAlternate = control.classList.contains('alternate');
+			expect(hasClassAlternate).toEqual(true);
+		});
+	});
+
+	describe('heading', () => {
+		it('should render heading if heading is set', async function () {
+			element.heading = 'header';
+			await elementUpdated(element);
+
+			const headerContent = element.shadowRoot?.querySelector('.header-content');
+			expect(headerContent).toBeTruthy();
+		});
+	});
+
+	describe('elevated', () => {
+		/**
+		 * @param scrollDist
+		 */
+		async function emulateScroll(scrollDist = 50) {
+			window['pageYOffset'] = scrollDist;
+			window.dispatchEvent(new CustomEvent('scroll'));
+			await elementUpdated(element);
+		}
+
+		beforeEach(function () {
+			element.fixed = true;
+		});
+
+		it('should default to no elevation with no scrolling', async function () {
```

Comment:
I can't see any relation to scrolling in the test itself.
Maybe this is more accurate:
```suggestion
		it('should default to no elevation', async function () {
```